tools/sigclient/pkg/query/concurrent_queries.go

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2021-2024 SigScalr, Inc.
+//
+// This file is part of SigLens Observability Solution
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package query
+
+import (
+	"fmt"
+	"sync"
+	"sync/atomic"
+
+	"github.com/fasthttp/websocket"
+	log "github.com/sirupsen/logrus"
+)
+
+var qid = int64(1)
+var connReset = int64(0)
+var timeout = int64(0)
+var failed = int64(0)
+var completed = int64(0)
```

Comment:
nit: it looks like these are counters for how many times they've happened. Rename them (e.g., `numFailed`). That makes your intention much more clear. When I first read this, I thought `timeout = 0` meant queries would never timeout